### PR TITLE
[Testing] Fix some build warnings

### PR DESCRIPTION
**Pull request by davezarzycki**

### CODE REVIEW

**Comment by davezarzycki** [comment]
@swift-ci please smoke test and merge

**Comment by davezarzycki** [comment]
@swift-ci please smoke test macos

File: stdlib/tools/swift-reflection-test/swift-reflection-test.c [hunk 1]
```diff
@@ -34,6 +34,10 @@
 #include <fcntl.h>
 #endif
 
+#if defined(__clang__) || defined(__GNUC__)
+#define NORETURN __attribute__((noreturn))
+#endif
```
Comment (mikeash) [inline stdlib/tools/swift-reflection-test/swift-reflection-test.c]: Do we need an `#else #define NORETURN` here for other compilers?

File: tools/swift-ast-script/swift-ast-script.cpp [hunk 2]
```diff
@@ -62,7 +62,8 @@ class Observer : public FrontendObserver {
 
 }
 
-int main(int argc, const char *argv[]) {
+// ISO C++ does not allow 'main' to be used by a program [-Wmain]
```
Comment (mikeash) [inline tools/swift-ast-script/swift-ast-script.cpp]: Huh, never knew about this.

File: stdlib/tools/swift-reflection-test/swift-reflection-test.c (hunk 1 above)
Comment (davezarzycki) [reply stdlib/tools/swift-reflection-test/swift-reflection-test.c]: Oh, duh.  Ya. Normally I'd not ifdef this, but given that Windows support is being pushed forward, I thought I'd make a nod towards portability. See also: #26868

File: tools/swift-ast-script/swift-ast-script.cpp (hunk 2 above)
Comment (davezarzycki) [reply tools/swift-ast-script/swift-ast-script.cpp]: Ya, clang top-of-tree is now warning about this by default.

**Pull request merged**